app/models/ems_refresh/save_inventory_helper.rb

```diff
@@ -1,14 +1,47 @@
 module EmsRefresh::SaveInventoryHelper
+  class TypedIndex
+    attr_accessor :record_index, :record_index_columns
+    attr_accessor :find_key
+    def initialize(records, find_key)
+      # Save the columns associated with the find keys, so we can coerce the
+      #   hash values during fetch
+      columns_hash = records.first.try(:class).try(:columns_hash_with_virtual)
+      record_index_columns = columns_hash.nil? ? [] : find_key.collect { |k| columns_hash[k.to_s] }
+
+      # Index the records by the values from the find_key
+      record_index = records.each_with_object({}) do |r, h|
+        h.store_path(find_key.collect { |k| r.send(k) }, r)
+      end
+
+      @find_key = find_key
+      @record_index = record_index
+      @record_index_columns = record_index_columns
```

Comment:
@kbrock why is the local variable needed instead of just initializing the ivar on line 9?